x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -347,6 +359,10 @@ private static boolean hasRemoteIndices(Collection<RoleDescriptor> roleDescripto
         return roleDescriptors != null && roleDescriptors.stream().anyMatch(RoleDescriptor::hasRemoteIndicesPrivileges);
     }
 
+    private static boolean hasWorkflowsRestriction(Collection<RoleDescriptor> roleDescriptors) {
+        return roleDescriptors != null && roleDescriptors.stream().anyMatch(RoleDescriptor::hasWorkflowsRestriction);
+    }
```

Comment:
It seems we need a check for: If workflow is used, the API key must have only a single role descriptor.

Reply:
Yes. I've forgot to add one. I'll fix that and add a test for it.